### PR TITLE
Support java AST String literal with desired encoding

### PR DESCRIPTION
**Pull request by winningsix**
This fixes https://github.com/rapidsai/cudf/issues/14397 to allow different encoding for string literal.
## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/ast/Literal.java [hunk 1]
```diff
@@ -205,7 +205,14 @@ public static Literal ofString(String value) {
     if (value == null) {
       return ofNull(DType.STRING);
     }
-    byte[] stringBytes = value.getBytes();
+    return ofString(value.getBytes());
```
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/ast/Literal.java:208]:
```suggestion
    return ofString(value.getBytes(StandardCharsets.UTF_8));
```

File: java/src/main/java/ai/rapids/cudf/ast/Literal.java [hunk 2]
```diff
@@ -205,7 +205,14 @@ public static Literal ofString(String value) {
...
+  }
+
+  /** Construct a string literal directly with byte array to skip transcosing. */
```
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/ast/Literal.java:211]:
```suggestion
  /** Construct a string literal directly with byte array to skip transcoding. */
```

File: java/src/main/java/ai/rapids/cudf/ast/Literal.java [hunk 3]
```diff
@@ -205,7 +205,14 @@ public static Literal ofString(String value) {
...
+  public static Literal ofString(byte[] stringBytes) {
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/ast/Literal.java:212]: nit: Might be nice to add UTF8 to the name of the method or in the comments.

**Review by revans2 — APPROVED** [review]

**Review by jlowe — APPROVED** [review]

**Comment by winningsix** [comment]
It seems failed test cases are not related. We only changed Java binding part. Rebase to have a further try.

**Pull request merged**